mos_tests/ironic/conftest.py

```diff
@@ -93,3 +95,92 @@ def baremetal_node(env_name, suffix):
     node.erase()
     disk.volume.erase()
     disk.delete()
+
+
+@pytest.yield_fixture
+def keypair(os_conn):
+    keypair = os_conn.create_key(key_name='ironic-key')
+    yield keypair
+    os_conn.delete_key(key_name=keypair.name)
+
+
+@pytest.yield_fixture
+def flavor(baremetal_node, os_conn):
+    flavor = os_conn.nova.flavors.create(name=baremetal_node.name,
+                                         ram=baremetal_node.memory,
+                                         vcpus=baremetal_node.vcpu,
+                                         disk=settings.IRONIC_DISK_GB)
+
+    yield flavor
+    flavor.delete()
+
+
+@pytest.yield_fixture(scope='module')
+def image_file():
+    image_file = common.gen_temp_file(prefix='image', suffix='img')
+    yield image_file
+    image_file.unlink(image_file.name)
+
+
+@pytest.yield_fixture
+def ubuntu_image(os_conn, image_file):
+    image = os_conn.glance.images.create(
+        name='ironic_trusty',
+        disk_format='raw',
+        container_format='bare',
+        hypervisor_type='baremetal',
+        cpu_arch='x86_64',
+        fuel_disk_info=json.dumps(settings.IRONIC_GLANCE_DISK_INFO))
```

Comment:
It would be better to add argument for ubuntu_image() of disk settings and take settings.IRONIC_GLANCE_DISK_INFO as default value.
This can be needed to extend tests with additional disk configurations.